src/providers/retryProvider.ts

```diff
@@ -260,11 +260,29 @@ export class RetryProvider extends ethers.providers.StaticJsonRpcProvider {
     return response;
   }
 
-  _trySend(provider: ethers.providers.StaticJsonRpcProvider, method: string, params: Array<unknown>): Promise<unknown> {
+  async _trySend(
+    provider: ethers.providers.StaticJsonRpcProvider,
+    method: string,
+    params: Array<unknown>
+  ): Promise<unknown> {
+    const loop = true;
     let promise = this._sendAndValidate(provider, method, params);
-    for (let i = 0; i < this.retries; i++) {
-      promise = promise.catch(() => delay(this.delay).then(() => this._sendAndValidate(provider, method, params)));
-    }
+    let i = 0;
+
+    do {
+      try {
+        await promise;
+        break;
+      } catch (err: unknown) {
+        if (++i >= this.retries) {
+          throw err;
+        }
+
+        await delay(this.delay);
+        promise = this._sendAndValidate(provider, method, params);
```

Comment:
Is this re-assignment necessary?

Reply:
It was given the surrounding code, but I was able to consolidate it in this commit: https://github.com/across-protocol/sdk/pull/761/commits/25516ed445a8246e621883efc40a32f43c4fe559

Reply:
Subsequently negated by this commit anyway: https://github.com/across-protocol/sdk/commit/d0a877df2808af58d5218f277dcfcc08b2da88d4